feat: cowswap volume fee

shoom3301 opened the pull request:
# Summary

Spec: https://www.notion.so/cownation/Launch-Volume-fee-on-Gnosis-Chain-94696716540f4e93a86fdfec56399735

0.1% fee should be added when:
 - Gnosis Chain
 - Not widget
 - No stable-stable trades

# To Test

Should satisfy the specification

anxolin reviewed (approved):
Very clean and easy to follow implementation

elena-zh reviewed (changes requested):
Hey @shoom3301 , great, thank you!
Changes LGTM in the widget, and in the app in all networks but Gnosis chain. 
Issues I've faced: 
1. No feature-flag
2. As far as I can understand from the spec, partner fee should not be applied when trade with [cherry-picked stable coins](https://www.notion.so/cownation/Launch-Volume-fee-on-Gnosis-Chain-94696716540f4e93a86fdfec56399735?pvs=4#4ba1e9f3fcb9458e914301a4d853f9dc). So when I pick
- SDAI/XDAI, SDAI/wXDAI, SDAI/EURE, SDAI/UDCS, SDAI/USDT
- XDAI/SDAI, XDAI/GBPe
- wxDAI/SDAI, wxDAI/GBPE
- EURe/SADI, EURe/GBPe
- GBPe/USDT, GBPe/USDC, GBPe/EURE, GBPe/wxDAI, GBPe/xDAI, GBPe/SDAI
- USDT/SDAI, USDT/GBPe
- USDC/SDAI, USDC/GBPe
partner fee is applied, however, should not.
![sdai-xdai](https://github.com/cowprotocol/cowswap/assets/70885163/8815e55b-b1a9-4870-ab44-b4e7ed4d24e0)

3. Also noticed that fee is not applied when I trade with different (not stabled from the cherry picked list) token pairs like HOP/1INCH, HOP/0xMR, WBTC/AGVE
![hop](https://github.com/cowprotocol/cowswap/assets/70885163/c9d41b2b-48a5-43d7-89ad-56224dfa9004)
![hop=1](https://github.com/cowprotocol/cowswap/assets/70885163/2277c484-b3ea-4c32-99a8-e8dec9585d49)
![wbtc-agve](https://github.com/cowprotocol/cowswap/assets/70885163/24122e23-1bcf-4b2f-b133-1a247172daf7)

shoom3301 commented:
@elena-zh thank you!
FIxed the logic and added [feature-flag](https://app.launchdarkly.com/projects/default/flags/isCowSwapFeeEnabled/targeting?env=production&env=barn&env=staging&env=test&env=develop&selected-env=test)

alfetopito commented on apps/cowswap-frontend/src/modules/volumeFee/state/cowswapFeeAtom.ts:
```diff
@@ -0,0 +1,42 @@
+
+const COWSWAP_VOLUME_FEE: VolumeFee = {
+  bps: 10, // 0.1%
+  recipient: '0x6b3214fD11dc91De14718DeE98Ef59bCbFcfB432', // Gnosis Chain Protocol fee safe
```
I know that PR comes after this one, but should afterwards change this to use the chain map, to make sure it's included only on gnosis chain based on the configuration, and make the logic simpler?

shoom3301 replied:
I love this suggestion, thanks!
https://github.com/cowprotocol/cowswap/pull/4547/commits/d6dd53dba7feffe422cd3d2fd9382305679dd2d9

alfetopito commented on apps/cowswap-frontend/src/modules/volumeFee/state/volumeFeeAtom.ts:
```diff
@@ -7,9 +7,25 @@ import { injectedWidgetPartnerFeeAtom } from 'modules/injectedWidget'
+  const cowSwapFee = get(cowSwapFeeAtom)
+  const widgetPartnerFee = get(widgetPartnerFeeAtom)
+
+  if (featureFlags.isCowSwapFeeEnabled) {
+    return cowSwapFee || widgetPartnerFee
+  }
```
Why return the `widgetPartnerFee` when `cowSwapFee` isn't defined?
If cowswap fee is enabled only that should be returned, no?

shoom3301 replied:
If `cowSwapFee` is not defined it might be a case when we in the widget mode, hence we should use widget partner fee

alfetopito reviewed (commented):
Maybe this PR should come after the one where the fee is more fine grained, to make it simpler to specify it per chain?

elena-zh reviewed (approved):
Hey @shoom3301 , now the changes LGTM!
I'm approving with assumption that some issues like incorrect fee percentate in a receipt, 'Receive' amount on wrap/etc. are fixed in #4502 PR. 
![10](https://github.com/cowprotocol/cowswap/assets/70885163/104116d0-56ab-4f84-bb5c-86832bc70a54)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/0cc10c59-743c-429a-bc32-474ea5336adc)
